agent/agent.go

```diff
@@ -34,26 +35,32 @@ type Agent struct {
 	cancelF context.CancelFunc
 	closing uint32
 	ctx     context.Context
-	sigs    chan os.Signal
```

Comment:
Why don't we want to handle signals in the `Agent`? It would seemingly make even _more_ sense to support signal handling in no-sandbox mode.

Reply:
We are handling signals. We weren't before. I'm just not using a channel to do it